gofn.go

```diff
@@ -107,15 +108,15 @@ func Run(ctx context.Context, buildOpts *provision.BuildOptions, containerOpts *
 	}(ctx, done)
 	select {
 	case <-ctx.Done():
-		log.Printf("trying to destroy container %v\n", ctx.Err())
+		log.Errorf("trying to destroy container %v\n", ctx.Err())
 	case <-done:
-		log.Println("trying to destroy container process done")
+		log.Debugln("trying to destroy container process done")
 	}
 	if machine != nil {
-		log.Printf("trying to delete machine ID:%v\n", machine.ID)
+		log.Debugf("trying to delete machine ID:%v\n", machine.ID)
 		deleteErr := buildOpts.Iaas.DeleteMachine(machine)
 		if deleteErr != nil {
-			log.Errorln("error trying to delete machine ", deleteErr)
+			err = fmt.Errorf("error trying to delete machine %v", deleteErr)
```

Comment:
Do we really want to override the container error on all these cases that you have change?

Reply:
initially I was only going to return the container error, but I found it interesting to use the previous message to facilitate debugging. What is better? Return the message from the container or add our log string?